packages/database/src/migrations/20220218015022-AddDataLakeDataElementsIntoTupaiaDataSource-modifies-data.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+import DATA_ELEMENT_CODES from './migrationData/20220218015022-AddDataLakeDataElementsIntoTupaiaDataSource-modifies-data/DataElementCodes - DataLake.json';
+import { generateId } from '../utilities/generateId';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  const formattedValues = DATA_ELEMENT_CODES.dataElementCodes
+    .map(code => `('${generateId()}', '${code}','dataElement','data-lake')`)
+    .join(',');
+  return db.runSql(`
+    INSERT INTO data_source (id, code, type, service_type)
+    VALUES
+    ${formattedValues}
+    ON CONFLICT ON CONSTRAINT data_source_code_type_key
+    DO UPDATE SET service_type = 'data-lake';
```

Comment:
One code (covid_n_test) has been found in `data-lake` as same as one of the data element code in tupaia. Have a chat with Juliana, and we decided to delete the survey it is associated to (done) and update its `service_type`.

Reply:
If this is being deleted we shouldn't need this right? I'm slightly worried that if another clash happened, this migration would move that data element to the data-lake with cleaning up the requisite question/survey. But I think I'm just being paranoid 😉 Should be fine 👍

Reply:
Emm that's right. Let me double check.

Reply:
Got another test to make sure there is not another clash. It is good to go. Very good thought!